### PR TITLE
Improve 'introduce variable for expression' when using tuples

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/39537

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs [hunk 1]
```diff
@@ -391,6 +389,160 @@ void M()
...
+@"
+using System;
+
+class C
+{
+    (string someString, int someInt) X() => default;
+
+    void M()
+    {
+        var (someString, someInt) = X();
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs:420]: ❔ When the user configures the use of explicit types everywhere, should this be `(string someString, int someInt) = X()`?

File: src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs [hunk 2]
```diff
@@ -391,6 +389,160 @@ void M()
...
+@"
+using System;
+
+class C
+{
+    (string, int) X() => default;
+
+    void M()
+    {
+        (string, int) {|Rename:value|} = X();
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs:544]: 💭 Another option is `var (item1, item2)`

File: src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs [hunk 3]
```diff
@@ -391,6 +389,160 @@ void M()
     {
         int {|Rename:v|} = 1 + 1;
     }
+}");
+        }
+
+        [Fact, WorkItem(39537, "https://github.com/dotnet/roslyn/issues/39537")]
+        public async Task IntroduceDeconstruction1()
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs:396]: ❔ Is this _always_ deconstructed, or are both options presented for the user to pick?

File: src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs:420]: yes.  good point.

File: src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs:544]: i particularly didn't like that.  :)  i felt the `item1`/`item2` would be worse than today :)

File: src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs:396]: always deconstructed.

File: src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs:515]: ok.  this is done.

File: src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs:857]: ok.  this is better now.  if it's going to be one of those names, we attempt to pick good local names based on type. e.g. `Customer customer`.

File: src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/IntroduceVariable/IntroduceLocalForExpressionTests.cs]: Now the user can pick between one or the other.  Looks like this:

image

**Review by sharwell — APPROVED** [review]

**Pull request merged**